### DISCUSSION TITLE
Allure .step and .attach is causing screenshots to link at end of the report while annotation step is adding screenshot below the step

### DISCUSSION THREAD

**Discussion by krishtoautomate** [allure-framework/allure2#2545] [Questions & Support] [answered]
### Describe the Bug

Allure .step and .attach is causing screenshots to link at end of the report while annotation step is adding screenshot below the step

### Steps to Reproduce

In java:
```
test.step("step 1");
Allure.step(message, Status.PASSED);
                    Allure.addAttachment("screenshot",  new ByteArrayInputStream(((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES)));

test.step("step 2");
Allure.step(message, Status.PASSED);
                    Allure.addAttachment("screenshot",  new ByteArrayInputStream(((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES)));
```

### Expected Behaviour

Every screenshot should attach under its respective step

### Screenshots or Additional Context

working only in annotations mode as expected when used @Step 

### What Language are you using?

Java

### What Framework/Allure Integration you are using?

allure-testng

### What version of Allure Integration you are using?

2.26.0

### What version of Allure Report you are using?

2.26.0

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by baev** [comment, marked as answer]
You can use the [attachments meta steps](https://github.com/allure-framework/allure2/pull/2107) feature. Wrap each attachment method with a lambda step, and make sure the step name is the same as the attachment.

```java
Allure.step("screenshot", () => {
    Allure.addAttachment(
        "screenshot",  
        new ByteArrayInputStream(((TakesScreenshot) driver).getScreenshotAs(OutputType.BYTES))
    );
});
```
